src/uct/cuda/base/cuda_md.c

```diff
@@ -55,6 +56,59 @@ UCS_PROFILE_FUNC(ucs_status_t, uct_cuda_base_detect_memory_type,
     return UCS_ERR_INVALID_ADDR;
 }
 
+static ucs_status_t uct_cuda_base_get_sys_dev(ucs_sys_device_t *sys_dev_p)
+{
+    CUdevice dev;
+    int attrib;
+
+    if (UCS_OK != UCT_CUDADRV_FUNC(cuCtxGetDevice(&dev))) {
+        return UCS_ERR_IO_ERROR;
+    }
+
+    if (UCS_OK != UCT_CUDADRV_FUNC(cuDeviceGetAttribute(&attrib,
+                    CU_DEVICE_ATTRIBUTE_PCI_DOMAIN_ID, dev))) {
+        return UCS_ERR_IO_ERROR;
+    }
+    sys_dev_p->bus_id.domain = (uint16_t)attrib;
+
+    if (UCS_OK != UCT_CUDADRV_FUNC(cuDeviceGetAttribute(&attrib,
+                    CU_DEVICE_ATTRIBUTE_PCI_BUS_ID, dev))) {
+        return UCS_ERR_IO_ERROR;
+    }
+    sys_dev_p->bus_id.bus = (uint8_t)attrib;
+
+    return UCS_OK;
+}
+
+ucs_status_t uct_cuda_base_mem_query(uct_md_h md, const void *addr,
+                                     size_t length,
+                                     uct_md_mem_attr_t *mem_attr_p)
+{
+    ucs_status_t status;
+
+    mem_attr_p->field_mask = 0;
+
+    if ((addr == NULL) || (length == 0)) {
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    status = uct_cuda_base_detect_memory_type(md, addr, length, &mem_attr_p->mem_type);
+    if (UCS_OK == status) {
+        mem_attr_p->field_mask |= UCT_MD_MEM_ATTR_FIELD_MEM_TYPE;
+    } else {
+        ucs_warn("unable to query mem_type for %p", (void*) addr);
```

Comment:
uct_cuda_base_detect_memory_type failed() need to fail this function, not just print a warning and continue
same for uct_cuda_base_get_sys_dev()

Reply:
If we change semantics in UCT API to caller setting field_mask bits then I agree we need to fail at different steps. Refer https://github.com/openucx/ucx/pull/4974#discussion_r407588420